src/main/java/uc/doc/downloaddoc/DownloadDocInteractor.java

```diff
@@ -0,0 +1,42 @@
+package uc.doc.downloaddoc;
+
+import java.time.LocalDateTime;
+
+import entities.StateTracker;
+import entities.User;
+
+public class DownloadDocInteractor implements DownloadDocInputBoundary {
+    private final DownloadDocFileAccessGateway downloadDocFileAccessGateway;
+    private final DownloadDocOutputBoundary downloadDocOutputBoundary;
+    private final StateTracker stateTracker;
+ 
+    public DownloadDocInteractor(
+        DownloadDocFileAccessGateway downloadDocFileAccessGateway, 
+        DownloadDocOutputBoundary downloadDocOutputBoundary, 
+        StateTracker stateTracker) 
+        {
+        this.downloadDocFileAccessGateway = downloadDocFileAccessGateway;
+        this.downloadDocOutputBoundary = downloadDocOutputBoundary;
+        this.stateTracker = stateTracker;
+    }
+
+    @Override
+    public DownloadDocResponseModel downloadDoc(DownloadDocRequestModel model) {
+        User user = stateTracker.getCurrentUser();
+        String documentId = model.getDocumentId();
+        String downloadPath;
+
+        if (!checkIfDocumentDownloaded(documentId)) {
```

Comment:
I think we have some miscommunication with regards to this use case. My understanding was that the use case was meant for downloading a document from the server to the user's local machine. This would mean that we do not need it be rendered in the view since it is a simple case of downloading a document, much like you would save an image from anywhere on the web to your local machine.

If it is a case of rendering a document from the server on the view, we should directly be using the remote_path as specified in local.properties along with the test_id/solution_id and no downloading to local would be required? This would be a different use case from what I was trying to build.

As for using a config variable for the local directory that files are downloaded to, I disagree with this as users should be allowed to choose where they want to save files to on their local machine.

For Line 29, I assumed checkIfDocumentDownloaded returns True if downloaded and False if not. Hence the if condition is checking for documents which have not already been downloaded.

Reply:
Is it possible for the view to render a PDF using only the remote path? My understanding of SFTP is that an SFTP request with credentials is necessary which would require that the file be downloaded locally. I am also unsure whether a remote path (using HTTP?) is possible with the PDF rendering library that we are using - @takedakento may know more.

The downloads are only for the rendering in the view and so it makes most sense to me that they are downloaded to the project directory (e.g. in the lib folder or a new 'data' folder). The motivation for this use case is that when a user clicks to view a test, we need to render the PDF for that test. If the path in the view model is null or empty, then this use case should be called to download the file to a local directory and then return the local path so that the view can render the file. Making the gateway return the local path I think is simplest as we don't want the view to look for a local file unless we know that it is downloaded.

Reply:
Also sorry for the miscommunication. I understand what you had in mind now and it is a perfect implementation of that.

Reply:
No worries, I should have clarified on my part before writing the code as well!